### ISSUE TITLE
On Mac, the kernel is looking in wrong place for libclingJupyter.{so,dylib,dll}

### ISSUE THREAD

**Issue by desmond-dsouza** [root-project/cling#133]
If I start `jupyter-notebook` from the _parent_ of <my_cling_dir>, the kernel starts and runs fine.

If I start it anywhere else, I get:
RuntimeError: Cannot find cling_2017-05-23_mac1011/lib/libclingJupyter.{so,dylib,dll}

Maybe it is looking for the dylib relative to the launch directory?

**Comment by Axel-Naumann** [comment]
Is this still the case? We had a couple of fixes to the "where's my libclingJupyter" logic lately (and I haven't heard any other complaints)...

**Comment by josephsieh** [comment]
Using cling version 0.5 installed from homebrew.  Then following https://github.com/root-project/cling/tree/master/tools/Jupyter instruction, and it is still the case that "where is my libclingJupyter" logic seems still does not work.

```
Traceback (most recent call last):
  File "/Volumes/Data/anaconda/bin/jupyter-cling-kernel", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/usr/local/Cellar/cling/0.5_1/libexec/share/cling/Jupyter/kernel/scripts/jupyter-cling-kernel", line 4, in <module>
    main()
  File "/usr/local/Cellar/cling/0.5_1/libexec/share/cling/Jupyter/kernel/clingkernel.py", line 344, in main
    ClingKernelApp.launch_instance()
  File "/Volumes/Data/anaconda/lib/python2.7/site-packages/traitlets/config/application.py", line 657, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-121>", line 2, in initialize
  File "/Volumes/Data/anaconda/lib/python2.7/site-packages/traitlets/config/application.py", line 87, in catch_config_error
    return method(app, *args, **kwargs)
  File "/Volumes/Data/anaconda/lib/python2.7/site-packages/ipykernel/kernelapp.py", line 457, in initialize
    self.init_kernel()
  File "/Volumes/Data/anaconda/lib/python2.7/site-packages/ipykernel/kernelapp.py", line 368, in init_kernel
    user_ns=self.user_ns,
  File "/Volumes/Data/anaconda/lib/python2.7/site-packages/traitlets/config/configurable.py", line 412, in instance
    inst = cls(*args, **kwargs)
  File "/usr/local/Cellar/cling/0.5_1/libexec/share/cling/Jupyter/kernel/clingkernel.py", line 128, in __init__
    raise RuntimeError('Cannot find ' + clingInstDir + '/lib/libclingJupyter.{so,dylib,dll}')
```

**Comment by haarc** [comment]
I have the same problem!

**Comment by Axel-Naumann** [comment]
Damn. Would someone mind taking a look at this and submitting a PR that fixes this, once and for all? Can't be *that* difficult ;-)

**Comment by Axel-Naumann** [comment]
Is this related to https://github.com/root-project/cling/issues/209 ?

**Comment by sherrycheryl** [comment]
I have the same problem!

That's what I did:

brew install cling
cling --version

pyenv version
pyenv virtualenv cpp
pyenv global cpp
pip install jupyter

pyenv activate cpp
jupyter notebook

brew list cling
export PATH=/usr/local/Cellar/cling/0.5_1/bin:$PATH
cd /usr/local/Cellar/cling/0.5_1/libexec/share/cling/Jupyter/kernel
pip3 install -e .
sudo jupyter-kernelspec install --user cling-cpp17
sudo jupyter-kernelspec install --user cling-cpp1z
sudo jupyter-kernelspec install --user cling-cpp14
sudo jupyter-kernelspec install --user cling-cpp11

Then I encountered:
Traceback (most recent call last):
  File "/Users/qinpeng/.pyenv/versions/cpp/bin/jupyter-cling-kernel", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/usr/local/Cellar/cling/0.5_1/libexec/share/cling/Jupyter/kernel/scripts/jupyter-cling-kernel", line 4, in <module>
    main()
  File "/usr/local/Cellar/cling/0.5_1/libexec/share/cling/Jupyter/kernel/clingkernel.py", line 344, in main
    ClingKernelApp.launch_instance()
  File "/Users/qinpeng/.pyenv/versions/3.6.5/envs/cpp/lib/python3.6/site-packages/traitlets/config/application.py", line 657, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-123>", line 2, in initialize
  File "/Users/qinpeng/.pyenv/versions/3.6.5/envs/cpp/lib/python3.6/site-packages/traitlets/config/application.py", line 87, in catch_config_error
    return method(app, *args, **kwargs)
  File "/Users/qinpeng/.pyenv/versions/3.6.5/envs/cpp/lib/python3.6/site-packages/ipykernel/kernelapp.py", line 465, in initialize
    self.init_kernel()
  File "/Users/qinpeng/.pyenv/versions/3.6.5/envs/cpp/lib/python3.6/site-packages/ipykernel/kernelapp.py", line 376, in init_kernel
    user_ns=self.user_ns,
  File "/Users/qinpeng/.pyenv/versions/3.6.5/envs/cpp/lib/python3.6/site-packages/traitlets/config/configurable.py", line 412, in instance
    inst = cls(*args, **kwargs)
  File "/usr/local/Cellar/cling/0.5_1/libexec/share/cling/Jupyter/kernel/clingkernel.py", line 128, in __init__
    raise RuntimeError('Cannot find ' + clingInstDir + '/lib/libclingJupyter.{so,dylib,dll}')
RuntimeError: Cannot find ../libexec/lib/libclingJupyter.{so,dylib,dll}

**Comment by AntoinePrv** [comment]
I wanted to PR on this but it seems to be solved in [master](https://github.com/root-project/cling/blob/ee374738072fab73d2b9896c968a000cf0a1bfb4/tools/Jupyter/kernel/clingkernel.py#L109) (pb from relative symlink not being resolve).

Perhaps you could do a bugfix release for mac users?

**Comment by Axel-Naumann** [comment]
If it was fixed in Feb 2019 it should have been fixed in cling 0.9. What do the xeus-cling people say on this, did you contact them?
